packages/vm/src/evm/interpreter.ts

```diff
@@ -109,6 +98,15 @@ export default class Interpreter {
     // Iterate through the given ops until something breaks or we hit STOP
     while (this._runState.programCounter < this._runState.code.length) {
       const opCode = this._runState.code[this._runState.programCounter]
+
+      if (
+        (opCode === 0x56 || opCode === 0x57 || opCode === 0x5e) &&
+        this._runState.validJumps.length === 0
```

Comment:
I'm not sure in what order JS evaluates an `A && B` statement, but if A is first evaluated and then B then I'd say swap A and B in this case, or probably even better cache a boolean which saves if we have ran JUMPDEST analysis.

Reply:
In fact I think we should use a cache, there is an edge case where there is a JUMP but there are no JUMPDESTs, in that case `validJumps.length` remains 0.

Reply:
Makes sense.  I've added a new property to `runState` to track this.